src/sinks/console.rs

```diff
@@ -51,22 +56,55 @@ impl crate::topology::config::SinkConfig for ConsoleSinkConfig {
 
         Ok((Box::new(sink), Box::new(future::ok(()))))
     }
+
+    fn input_type(&self) -> DataType {
+        DataType::Any
+    }
 }
 
 fn encode_event(event: Event, encoding: &Option<Encoding>) -> Result<String, ()> {
-    let log = event.into_log();
+    match event {
+        Event::Log(log) => {
+            if (log.is_structured() && encoding != &Some(Encoding::Text))
+                || encoding == &Some(Encoding::Json)
+            {
+                let bytes = serde_json::to_vec(&log.all_fields())
+                    .map_err(|e| panic!("Error encoding: {}", e))?;
+                String::from_utf8(bytes)
+                    .map_err(|e| panic!("Unable to convert json to utf8: {}", e))
+            } else {
+                let s = log
+                    .get(&event::MESSAGE)
+                    .map(|v| v.to_string_lossy())
+                    .unwrap_or_else(|| "".into());
+                Ok(s)
+            }
+        }
+        Event::Metric(metric) => serde_json::to_string(&metric).map_err(|_| ()),
```

Comment:
do we want to attempt to just `fmt::Display` the type if encoding json is not provided?

Reply:
We'd have to implement `Display` for `Metric` and I'm not sure that's super worthwhile. My thought process was that we already default to JSON for structured data and metrics are structured data, but also that it's not a huge deal.